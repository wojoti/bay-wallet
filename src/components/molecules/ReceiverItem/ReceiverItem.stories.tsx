import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {ReceiverItem} from 'components/molecules';

export default {
  title: 'Molecules/ReceiverItem',
  component: ReceiverItem,
  argTypes: {
    onPress: {action: 'onPress'},
  },
} as ComponentMeta<typeof ReceiverItem>;
const Template: ComponentStory<typeof ReceiverItem> = args => (
  <ReceiverItem {...args} />
);

export const Example = Template.bind({});
Example.args = {
  testId: '',
  id: '0',
  firstname: 'First Name',
  selectedId: '1',
};

export const Selected = Template.bind({});
Selected.args = {
  testId: '',
  id: '0',
  firstname: 'First Name',
  selectedId: '0',
};
