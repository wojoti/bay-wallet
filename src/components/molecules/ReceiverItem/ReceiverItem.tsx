import {
  ComposableButton,
  Container,
  Img,
  ImgType,
  Label,
} from 'components/atoms';
import {PropsWithChildren} from 'react';
import style from './ReceiverItem.style';

export type ReceiverItemProps = PropsWithChildren<{
  id: string;
  firstname: string;
  onPress: (id: string) => void;
  selectedId: string;
  testId?: string;
}>;

const ReceiverItem = ({
  id,
  firstname,
  onPress,
  selectedId,
  testId,
}: ReceiverItemProps) => {
  const onReceiverPress = () => {
    onPress(id);
  };
  return (
    <Container customStyle={style.wrapper} testId={testId} align="flex-start">
      {id === selectedId ? (
        <ComposableButton
          customStyle={style.buttonPressed}
          onPress={onReceiverPress}
          testId="receiveritem-button">
          <Img
            testId="receiveritem-img"
            src={ImgType.iconUser}
            height={24}
            width={23}
            alt={''}
          />
        </ComposableButton>
      ) : (
        <ComposableButton
          customStyle={style.button}
          onPress={onReceiverPress}
          testId="receiveritem-button">
          <Img
            testId="receiveritem-img"
            src={ImgType.iconChartSelected}
            height={24}
            width={23}
            alt={''}
          />
        </ComposableButton>
      )}

      <Container customStyle={style.spacing}>
        <Label testId="receiveritem-label">{firstname}</Label>
      </Container>
    </Container>
  );
};

export default ReceiverItem;
