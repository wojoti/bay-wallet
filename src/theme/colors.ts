export const lightPrimary = '#4E439E';
export const primary = '#363062';
export const secondary = 'white';

export const darkGrey = '#595959';
export const grey = '#6B6B6B';
export const vlightGrey = '#D1D1D1';
export const borderGrey = '#E2E2E2';

export const iconGrey = '#BDBDBD';

export const searchGrey = '#F5F5F5';

export const separator = '#F6F6F6';

export const green = '#61C277';
export const lime = '#53E520';
export const red = '#E52020';

export const successGreen = '#3EB290';

export const pink = '#D8B9C3';
export const darkPurple = '#26204E';
export const balanceWhite = '#F8F8FF';

export const marketDarkGrey = '#3D454B';

export const cardYellow = '#FFEF60';

export const buttonGrey = '#EFEFEF';
