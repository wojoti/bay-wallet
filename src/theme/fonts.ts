export const font = 'Poppins-Light';
export const boldFont = 'Poppins-Bold';
