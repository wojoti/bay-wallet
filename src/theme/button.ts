import {StyleSheet} from 'react-native';
import {primary, secondary} from './colors';
const buttonStyle = StyleSheet.create({
  primary: {
    borderWidth: 0,
    backgroundColor: primary,
  },
  secondary: {
    borderColor: primary,
    backgroundColor: secondary,
  },
});

export const buttonTextStyle = StyleSheet.create({
  primary: {
    color: secondary,
  },
  secondary: {
    color: primary,
  },
});

export default buttonStyle;
