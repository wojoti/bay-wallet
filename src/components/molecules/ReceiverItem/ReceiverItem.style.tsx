import {StyleSheet} from 'react-native';
import {buttonGrey, primary} from 'theme/colors';

const style = StyleSheet.create({
  wrapper: {
    width: 'auto',
    marginRight: 20,
  },
  button: {
    backgroundColor: buttonGrey,
    borderRadius: 8,
    width: 60,
    height: 60,
  },
  buttonPressed: {
    backgroundColor: primary,
    borderRadius: 8,
    width: 60,
    height: 60,
  },
  spacing: {
    marginVertical: 10,
  },
});
export default style;
