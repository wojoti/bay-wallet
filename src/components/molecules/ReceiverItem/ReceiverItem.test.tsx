import '@testing-library/jest-native/extend-expect';
import {render, screen, within} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import {buttonGrey, primary} from 'theme/colors';
import ReceiverItem, {ReceiverItemProps} from './ReceiverItem';

const mockOnPress = jest.fn();
const props: ReceiverItemProps = {
  testId: 'test-receiveritem-id',
  id: '0',
  firstname: 'First Name',
  selectedId: '1',
  onPress: mockOnPress,
};
const additionalProps: ReceiverItemProps = {
  ...props,
  selectedId: '0',
};
test('should match snapshot', () => {
  const tree = TestRenderer.create(<ReceiverItem {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render ReceiverItem', () => {
  render(<ReceiverItem {...props} />);
  const ReceiverItemElement = screen.getByTestId('test-receiveritem-id');
  expect(ReceiverItemElement).toBeOnTheScreen();
});

test('should render ReceiverItem - button', () => {
  render(<ReceiverItem {...props} />);
  const ReceiverItemElement = screen.getByTestId('test-receiveritem-id');
  expect(ReceiverItemElement).toBeOnTheScreen();
  const btnElement = within(ReceiverItemElement).getByTestId(
    'receiveritem-button',
  );
  expect(btnElement).toBeOnTheScreen();
});

test('should render ReceiverItem - button - style', () => {
  render(<ReceiverItem {...props} />);
  const ReceiverItemElement = screen.getByTestId('test-receiveritem-id');
  expect(ReceiverItemElement).toBeOnTheScreen();
  const btnElement = within(ReceiverItemElement).getByTestId(
    'receiveritem-button',
  );
  expect(btnElement).toBeOnTheScreen();
  expect(btnElement).toHaveStyle({backgroundColor: buttonGrey});
});
test('should render ReceiverItem - button - pressed style', () => {
  render(<ReceiverItem {...additionalProps} />);
  const ReceiverItemElement = screen.getByTestId('test-receiveritem-id');
  expect(ReceiverItemElement).toBeOnTheScreen();
  const btnElement = within(ReceiverItemElement).getByTestId(
    'receiveritem-button',
  );
  expect(btnElement).toBeOnTheScreen();
  expect(btnElement).toHaveStyle({backgroundColor: primary});
});

test('should render ReceiverItem - button - img', () => {
  render(<ReceiverItem {...props} />);
  const ReceiverItemElement = screen.getByTestId('test-receiveritem-id');
  expect(ReceiverItemElement).toBeOnTheScreen();
  const btnElement = within(ReceiverItemElement).getByTestId(
    'receiveritem-button',
  );
  expect(btnElement).toBeOnTheScreen();
  const imgElement = within(btnElement).getByTestId('receiveritem-img');
  expect(imgElement).toBeOnTheScreen();
});

test('should render ReceiverItem - label', () => {
  render(<ReceiverItem {...props} />);
  const ReceiverItemElement = screen.getByTestId('test-receiveritem-id');
  expect(ReceiverItemElement).toBeOnTheScreen();
  const labelElement =
    within(ReceiverItemElement).getByTestId('receiveritem-label');
  expect(labelElement).toBeOnTheScreen();
});
