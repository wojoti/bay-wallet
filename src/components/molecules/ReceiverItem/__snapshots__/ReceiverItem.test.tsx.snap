// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "flex-start",
        "flex": undefined,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "width": "100%",
      },
      {
        "marginRight": 20,
        "width": "auto",
      },
    ]
  }
  testID="test-receiveritem-id"
>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderStyle": "solid",
          "justifyContent": "center",
          "opacity": 1,
        },
        false,
        {
          "backgroundColor": "#EFEFEF",
          "borderRadius": 8,
          "height": 60,
          "width": 60,
        },
      ]
    }
    testID="receiveritem-button"
  >
    <SvgMock
      height={24}
      testID="receiveritem-img"
      width={23}
    />
  </View>
  <View
    style={
      [
        {
          "alignItems": "stretch",
          "flex": undefined,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        {
          "marginVertical": 10,
        },
      ]
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontFamily": "Poppins-Light",
          "fontSize": 14,
          "fontWeight": "400",
          "textAlign": "left",
        }
      }
      testID="receiveritem-label"
    >
      First Name
    </Text>
  </View>
</View>
`;
